### DISCUSSION TITLE
Changing storage disk for uploaded files

### DISCUSSION THREAD

**Discussion by eriktobben** [livewire/livewire#8397] [1. Help] [answered]
Hi!

In the documentation it shows the following code to store a uploaded file to another disk:
` $this->photo->store(path: 'photos', 's3');`

But when trying that I get the following error: `Cannot use positional argument after named argument.`

I have tried adding `disk` or `storage` before s3, but that is not correct.

What is the correct way of specifying a disk for file uploads in Livewire?

**Comment by joshhanley** [comment, marked as answer]
@eriktobben just had a look and that store method is a Laravel `UploadedFile` method so the parameter name is techncially `options` as under the hood it gets converted to a disk option.

So shorthand would be this:

```php
$this->photo->store(path: 'photos', options: 's3');
```

Long hand (and probably the way I would prefer as it makes more sense):

```php
$this->photo->store(path: 'photos', options: ['disk' => 's3']);
```

**Reply by ifet4u** [reply]
Thanks  man.
